packages/react/avatar/src/Avatar.tsx

```diff
@@ -54,25 +54,55 @@ const IMAGE_NAME = 'AvatarImage';
 
 type AvatarImageElement = React.ElementRef<typeof Primitive.img>;
 type PrimitiveImageProps = React.ComponentPropsWithoutRef<typeof Primitive.img>;
+
 interface AvatarImageProps extends PrimitiveImageProps {
   onLoadingStatusChange?: (status: ImageLoadingStatus) => void;
 }
 
 const AvatarImage = React.forwardRef<AvatarImageElement, AvatarImageProps>(
   (props: ScopedProps<AvatarImageProps>, forwardedRef) => {
     const { __scopeAvatar, src, onLoadingStatusChange = () => {}, ...imageProps } = props;
+
     const context = useAvatarContext(IMAGE_NAME, __scopeAvatar);
-    const imageLoadingStatus = useImageLoadingStatus(src, imageProps.referrerPolicy);
+    const imageLoadingStatus = useImageLoadingStatus(src, props.asChild, imageProps.referrerPolicy);
     const handleLoadingStatusChange = useCallbackRef((status: ImageLoadingStatus) => {
       onLoadingStatusChange(status);
       context.onImageLoadingStatusChange(status);
     });
 
     useLayoutEffect(() => {
-      if (imageLoadingStatus !== 'idle') {
-        handleLoadingStatusChange(imageLoadingStatus);
+      if (!src) {
+        handleLoadingStatusChange('error');
+        return;
       }
-    }, [imageLoadingStatus, handleLoadingStatusChange]);
+
+      handleLoadingStatusChange('loading');
+    }, [handleLoadingStatusChange, src]);
+
+    if (props.asChild && props.children) {
+      // Ensure children is a valid React element
+      const child = React.Children.only(props.children) as React.ReactElement;
+
+      const { asChild, children, ...restProps } = props;
+
+      const childProps = child.props;
+
+      // Clone the child to add onLoad and onError event listeners
+      return React.cloneElement(child, {
+        ...restProps,
+        ...child.props,
+        onError: (event: React.SyntheticEvent<HTMLImageElement, Event>) => {
+          console.log('error');
+          handleLoadingStatusChange('error');
+          if (childProps.onError) childProps.onError(event);
```

Comment:
question: @chaance do we also need to trigger `onError` callback from `props`?

Reply:
For me make sense, considering that the child component might have some implementation of it 🤔.

But Chance can give us a better overview if it's necessary or not

Reply:
IMO, to mimic the natural CSS flow, we should check and send the event to both the child's and parent's callback, so that if any using the parent callback the code won't break thus avoiding chaos.